.githooks/pre-commit

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+STASH_NAME="pre-commit-$(date +%s)"
+git stash save -q --keep-index $STASH_NAME
+
+CHANGED_DTHS=$(git diff --cached --name-only | grep .*.groovy)
+ERROR_COUNT=0
+# echo $CHANGED_DTHS
+while IFS= read -r DTH; do
+  echo "Verifying $DTH"
+  ERRORS=$(groovyc $DTH 2>&1 | grep ".groovy:")
+  # echo $ERRORS
+  IMPORTANT_ERRORS=$(echo $ERRORS | grep -v "unable")
+  if [[ ${#IMPORTANT_ERRORS} -eq 0 ]]; then
+    echo "No disqualifying compilation errors found"
+  else
+    echo "$DTH failed to compile, run groovyc on your source file for the full error: $ERRORS"
+    ERROR_COUNT=$((ERROR_COUNT + 1))
+  fi
+  echo "======================================================================="
+done < <(git diff --cached --name-only | grep .*.groovy)
+
+STASHES=$(git stash list)
```

Comment:
I the stash logic must be left over and doesn't do anything because:
* == will never be true because $STASHES is a list, and each element is annotated with more than just the stash name
* we shouldn't need to stash pop anyway because stash save --keep-index didn't actually stash our changes

So in the end, can't we get rid of the stash logic altogether?

Reply:
It was working when I tested it, but I can take it out since we're not _really_ compiling anything.

Reply:
OK, I see what's being done here.  However, I think the substring match with == is going be pretty fragile.  This is notoriously unportable.  It definitely does not work in my environment.

If you can get $STASHES to only contain the thing you're looking for, this will be much more likely to work everywhere.

I got this working in my environment with the following:

`STASHES=$(git stash list -n 1 --format=%s | cut -d: -f2 | cut -d\  -f2)
if [[ $STASHES == $STASH_NAME" ]]; then
  git stash pop -q
fi`

Reply:
I assume you meant `if [[ $STASHES == "$STASH_NAME" ]]; then` and not `if [[ $STASHES == $STASH_NAME" ]]; then`?